### Задание 7.1

Даны две независимые выборки. Не соблюдается условие нормальности

x1 380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

### Решение
Так как выборок две и они независимы, применяем критерий Манна Уитни. Вывод: статистически значимых различий не обнаружено на уровне значимости $\alpha$ = 0.05

In [1]:
import numpy as np
import scipy.stats as stats

x1 = np.array([380,420, 290])
y1 = np.array([140,360,200,900])

stats.mannwhitneyu(x1, y1)


MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

### Задание 7.2

Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150, 130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

### Решение
Так как выборок три и они зависимы, применяем критерий Фридмана. Вывод: различия между выборками статистически значимые на уровне статистической значимости $\alpha$=0.05. Препарат работает.

In [2]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

### Задача 7.3
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

### Решение
Так как выборок две и они зависимы, применяем критерий Уилкоксона. Вывод: статистически значимых различий не обнаружено на уровне значимости $\alpha$ = 0.05. Препарат не работает.


In [4]:
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])

stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

### Задача 7.4
Даны 3 группы учеников плавания.

В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

### Решение
Так как выборок три и они независимы, испольщуем критерий Краскела – Уоллиса. Вывод: статистически значимых различий в группах не обнаружено на уровне значимости $\alpha$ = 0.05

In [5]:
x1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(x1, x2, x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

### Задача 7.5
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

### Решение
Так как выборка подчинена нормальному распределению и неизвестна дисперсия, используем критерий Стьюдента.


$$T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}},$$

In [9]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

#mean = X.mean()
def mean(x):
    return sum(x)/len(x)

#std = X.std(ddof=1)
def std(x):
     m = mean(x)
     return np.sqrt(sum(map(lambda t: (t - m)**2, x)) / (len(x) - 1))

t_fact = (mean(X) - 2.5) / std(X) * np.sqrt(len(X))
t_fact

0.563061366180296

По таблице для критерия Стьюдента находим t = 1.833 (для $\alpha$ = 0.05 и 9 степенями свободы). Так как t > t_fact (1.833 > 0.56), то гипотеза верна и изделия соответствуют заявленному размеру. 